### PR TITLE
Decouple Web lib from UI

### PR DESCRIPTION
**Pull request by Ponant**
Fixes https://github.com/AzureAD/microsoft-identity-web/issues/1034

### CODE REVIEW

File: src/Microsoft.Identity.Web/AzureADB2COpenIDConnectEventHandlers.cs [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Task OnRemoteFailure(RemoteFailureContext context)
             {
                 _errorAccessor.SetMessage(context.HttpContext, message);
 
-                context.Response.Redirect($"{context.Request.PathBase}/MicrosoftIdentity/Account/Error");
+                context.Response.Redirect($"{context.Request.PathBase}{Options.ErrorPath}");
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/AzureADB2COpenIDConnectEventHandlers.cs:95]: same thing.

File: src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs [hunk 2]
```diff
@@ -133,5 +134,19 @@ internal bool HasClientCredentials
         /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
         /// </summary>
         public string? UserAssignedManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Sets the ResetPassword route path.
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs:139]:
```suggestion
        /// ResetPassword path within the application's base path.
```

(proposing to remove the 'Sets' as it's set/get. Adding precision on what the path is.

File: src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs [hunk 3]
```diff
@@ -133,5 +134,19 @@ internal bool HasClientCredentials
...
+        /// Defaults to /MicrosoftIdentity/Account/ResetPassword,
+        /// which is the value used by Microsoft.Identity.Web.UI.
+        /// </summary>
+        public PathString ResetPasswordPath { get; set; } = new PathString("/MicrosoftIdentity/Account/ResetPassword");
+
+        /// <summary>
+        /// Sets the Error route path.
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs:146]:
```suggestion
        /// Error path within the application's base path.
```

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @Ponant 

Great improvement!

**Comment by Ponant** [comment]
You are welcome @jmprieur.
Notice that the only thing I did not attempt to change is the 
````cshsharp
internal const string BlazorChallengeUri = "MicrosoftIdentity/Account/Challenge?redirectUri=";
```` 
Which lives in the Web library. So I believe it will be great to find a way to have `MicrosoftIdentity/Account/Challenge` set by the user as part of the configuration, so any suggestion for a future PR are welcomed.

File: src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs:139]: @Ponant are you able to update the comment as per @jmprieur 's suggestion? thx.

**Comment by Ponant** [comment]
@jennyf19 , done

**Comment by jennyf19** [comment]
@Ponant can you remove the Microsoft.Identity.Web.xml file from the PR? I'll then merge to another branch so i can pull it locally and test.

**Comment by Ponant** [comment]
@jennyf19 , is it OK now?

**Comment by jennyf19** [comment]
sorry, i meant just not push it up, so there is no conflict. sorry about the confusion. i'll resolve it.

**Comment by jennyf19** [comment]
@Ponant in `MicrosoftIdentityConsentAndConditionalAccessHandler` you can get the IOptionsMonitor and the Scheme from the ServiceProvider, which is passed into the constructor already, and then do something [like this](https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/TokenAcquisition.cs#L610).

do you want to propose a new PR?

**Comment by Ponant** [comment]
@jennyf19 , I can look into that when I have time, just giv eit some time. How is this url triggered? I guess I need a blazor wasm to invoke this api, but at which stage? Obvisouly, I did not dig into that but I can look at it during weekends, I do not expect it to be difficult.

**Pull request merged**